### DISCUSSION TITLE
Unable to restore an deleted instance or volume from backup

### DISCUSSION THREAD

**Discussion by shaerul** [apache/cloudstack#10143] [Q&A] [answered]
I am using the backup and restore feature in ACS 4.20 for the first time. I believed that a deleted instance could be recovered if I backed it up to an NAS using the NFS protocol.

To test this, I backed up an instance, deleted it, and then tried to restore it by clicking "Restore Instance Backup." However, I received an error message:
"Restore Instance Backup (backup): Unable to restore VM with the current backup as the backup has a different number of disks compared to the VM."

Next, I tried "Restore and Attach Backup Volume" by selecting the volume and attaching it to another existing instance. This attempt also failed with the error:
"Cannot invoke 'com.cloud.storage.VolumeVO.getPoolId()'."

Does anyone know what might have gone wrong? I couldn't find any documentation that explains or supports this process.

**Comment by rajujith** [comment, marked as answer]
@shaerul  Restoring a deleted/Expunged instance is not supported in the B&R framework; the same applies to the NAS provider. However, this should be improved in an upcoming release. Here is the PR in progress: https://github.com/apache/cloudstack/pull/10140.
